src/renderers/WebGLRenderer.js

```diff
@@ -2091,6 +2091,8 @@ class WebGLRenderer {
 
 						const group = groups[ i ];
 
+						if ( group.uniforms.length === 0 ) continue;
```

Comment:
I understand you have added this line to prevent a processing from a uniforms group without defined uniforms.

But is this a real use case? Why should a developer setup is instance of `UniformsGroup` like that?

Reply:
Also regarding UBOs errors are sometimes misleading.

For example if you forget to bind it in JS it will throw the error `GL_INVALID_OPERATION: It is undefined behaviours to use a uniform buffer that is too small.`.

Maybe we can add another PR in the future just to help developers to make it easier to understand.